client/transaction-pool/src/testing/pool.rs

```diff
@@ -219,10 +219,9 @@ fn should_revalidate_during_maintenance() {
 
 	pool.api.push_block(1, vec![xt1.clone()]);
 
-	notifier.clear();
 	block_on(pool.maintain(block_event(1)));
 	assert_eq!(pool.status().ready, 1);
-	block_on(notifier.next());
+	block_on(notifier.next_blocking());
```

Comment:
What is the difference between next and next_blocking?

Reply:
`next_blocking` first clears the queue from all previous timer events, then starts to wait next one (same as `clear()` + `next()` before)

Reply:
Hmmm. If that is the case, what did change to the version before than? I don't see any difference.

Reply:
Now it clears events after `maintain`, so if during `maintain` the timer is fired, it is not handled before maintain is finished 😅

Reply:
Ahh, so we just than wait one iteration more?

Reply:
yes, one iteration that is guaranteed to be handled by timer handler